### PR TITLE
Sequence tree map cleanup

### PR DESCRIPTION
**Pull request by dbolotin**

### CODE REVIEW

File: src/main/java/com/milaboratory/core/clustering/Clustering.java [hunk 1]
```diff
@@ -87,16 +112,21 @@ public boolean isFinished() {
 
             Node<T[]> current;
 
+            // Used on each iteration to prevent double processing of the same trie node (NeighborhoodIterator may
+            // return the same node several times)
+            // Used as IdentityHashMap ...
             final HashSet<Node<T[]>> processedNodes = new HashSet<>();
```
Comment (dbolotin) [inline src/main/java/com/milaboratory/core/clustering/Clustering.java]: @PoslavskySV Why don't we use `IdentityHashMap` here?

Comment (PoslavskySV) [reply src/main/java/com/milaboratory/core/clustering/Clustering.java]: in fact it is not important, both will give the same performance

Comment (dbolotin) [reply src/main/java/com/milaboratory/core/clustering/Clustering.java]: Exactly 😿

Comment (dbolotin) [reply src/main/java/com/milaboratory/core/clustering/Clustering.java]: There is no such class as IdentityHash**Set**, actually.

**Pull request merged**